feat: customize atlantis.yaml file name in server side config (#2798)
runatlantis/atlantis: runatlantis.io/docs/server-side-repo-config.md

Firstly, you need to deploy 2 Atlantis servers, `production-server` and `staging-server`.
Each servers have different permissions and different `repos.yaml` file.
`repos.yaml` contains `repo_config_file` key to specify the repository config file name.

```yaml
# repos.yaml
repos:
- id: /.*/
  repo_config_file: atlantis-production.yaml # for production-server
  # repo_config_file: atlantis-staging.yaml # for staging-serever
```

Then, you need to create `atlantis-production.yaml` and `atlantis-staging.yaml` files in the repository.
See the configuration examples in [atlantis.yaml](repo-level-atlantis-yaml.html).

```yaml
# atlantis-production.yaml
version: 3
projects:
- name: project
  branch: /production/
  dir: infrastructure/production
---
# atlantis-staging.yaml
version: 3
projects:
  - name: project
    branch: /staging/
    dir: infrastructure/staging
```

Now, you setup 2 webhook URLs for your repository, which send events to `production-server` and `staging-server` respectively.
Each servers handle different repository config files.

:::tip Notes

> * If `no projects` comments are annoying, set [--silence-no-projects](server-configuration.html#silence-no-projects).

Author: ```suggestion
* If `no projects` comments are annoying, set [--silence-no-projects](server-configuration.html#silence-no-projects).
* You can customize command trigger executable name from `atlantis` to something you want by setting [Executable Name](server-configuration.html#executable-name).
```

> * When using different atlantis server vcs users such as `@atlantis-staging`, you can call `@atlantis-staging plan` instead `atlantis plan` to call `stating-server` only.

Author: I didn't know this usage, but we can call one of atlantis-server by using vcs user.

https://github.com/runatlantis/atlantis/blob/585ce1e888f94c83a8774e56a9ef310c1d6066d7/server/events/comment_parser.go#L98-L109

Reviewer: That is a nice call out, and something I learned today. @krrrr38 Are you asking a question about `run apply` or pointing out that its dangerous and useless? I think it should be removed, but we don't know any other use cases by users. It seems to just be an alias for atlantis. 

https://github.com/runatlantis/atlantis/blob/585ce1e888f94c83a8774e56a9ef310c1d6066d7/server/events/comment_parser.go#L147-L150

Author: @GenPage 

I discuss it in the [other PR, too](https://github.com/runatlantis/atlantis/pull/2808#discussion_r1051501348). I'm worry about that almost users don't know it and it might execute unintentional apply.

But currently this feature is already provided and we might be better to discuss it in the next major release and so on, not in this PR.

See [Custom Workflows](custom-workflows.html) for more details on writing
custom workflows.

### Multiple Atlantis Servers Handle The Same Repository

> Sometimes, you want to run multiple Atlantis servers handles a repository for reasons such as the server's permissions.

Reviewer 2: ```suggestion
Running multiple Atlantis servers to handles the same repository can be done to separate permissions for each Atlantis server.
```

> In this case, you can use different [atlantis.yaml](repo-level-atlantis-yaml.html) repository config file by using different `repos.yaml`.

Reviewer 2: Best to drop the word `you`

```suggestion
In this case, a different [atlantis.yaml](repo-level-atlantis-yaml.html) repository config file can be used by using different `repos.yaml` files.
```

> For example, let's consider a situation that you want to separate `production-server` handles `atlantis-production.yaml` and `staging-server` handles `atlantis-staging.yaml`.

Reviewer 2: ```suggestion
For example, consider a situation where a separate `production-server` atlantis uses repo config `atlantis-production.yaml` and `staging-server` atlantis uses repo config `atlantis-staging.yaml`.
```

> Firstly, you need to deploy 2 Atlantis servers, `production-server` and `staging-server`.

Reviewer 2: ```suggestion
Firstly, deploy 2 Atlantis servers, `production-server` and `staging-server`.
```

> Each servers have different permissions and different `repos.yaml` file.

Reviewer 2: ```suggestion
Each server has different permissions and a different `repos.yaml` file.
```

> `repos.yaml` contains `repo_config_file` key to specify the repository config file name.

Reviewer 2: ```suggestion
The `repos.yaml` contains `repo_config_file` key to specify the repository atlantis config file path.
```

```yaml
# repos.yaml
repos:
- id: /.*/

>   repo_config_file: atlantis-production.yaml # for production-server

Reviewer 2: ```suggestion
  # for production-server
  repo_config_file: atlantis-production.yaml
```

>   # repo_config_file: atlantis-staging.yaml # for staging-serever

Reviewer 2: ```suggestion
  # for staging-server
  # repo_config_file: atlantis-staging.yaml
```

```

> Then, you need to create `atlantis-production.yaml` and `atlantis-staging.yaml` files in the repository.

Reviewer 2: ```suggestion
Then, create `atlantis-production.yaml` and `atlantis-staging.yaml` files in the repository.
```

See the configuration examples in [atlantis.yaml](repo-level-atlantis-yaml.html).

```yaml
# atlantis-production.yaml
version: 3
projects:
- name: project
  branch: /production/
  dir: infrastructure/production
---
# atlantis-staging.yaml
version: 3
projects:
  - name: project
    branch: /staging/
    dir: infrastructure/staging
```

> Now, you setup 2 webhook URLs for your repository, which send events to `production-server` and `staging-server` respectively.

Reviewer 2: ```suggestion
Now, 2 webhook URLs can be setup for the repository, which send events to `production-server` and `staging-server` respectively.
```

Each servers handle different repository config files.

:::tip Notes
* If `no projects` comments are annoying, set [--silence-no-projects](server-configuration.html#silence-no-projects).

> * You can customize command trigger executable name from `atlantis` to something you want by setting [Executable Name](server-configuration.html#executable-name).

Reviewer 2: ```suggestion
* The command trigger executable name can be reconfigured from `atlantis` to something else by setting [Executable Name](server-configuration.html#executable-name).
```

> * When using different atlantis server vcs users such as `@atlantis-staging`, you can call `@atlantis-staging plan` instead `atlantis plan` to call `staging-server` only.

Reviewer 2: ```suggestion
* When using different atlantis server vcs users such as `@atlantis-staging`, the comment `@atlantis-staging plan` can be used instead `atlantis plan` to call `staging-server` only.
```

See [Custom Workflows](custom-workflows.html) for more details on writing
custom workflows.

### Multiple Atlantis Servers Handle The Same Repository

> Running multiple Atlantis servers to handles the same repository can be done to separate permissions for each Atlantis server.

Reviewer 2: ```suggestion
Running multiple Atlantis servers to handle the same repository can be done to separate permissions for each Atlantis server.
```
